### PR TITLE
plugins/bufferline: fix deprecated highlight options

### PR DESCRIPTION
**Pull request by GaetanLepage**
Fixes https://github.com/pta2002/nixvim/issues/292

### CODE REVIEW

File: plugins/bufferlines/bufferline.nix [hunk 1]
```diff
@@ -11,9 +11,15 @@ with lib; let
...
+    fg = helpers.mkNullOrOption types.str "foreground color";
 
-    guibg = helpers.mkNullOrOption types.str "background color";
+    bg = helpers.mkNullOrOption types.str "background color";
+
+    sp = helpers.mkNullOrOption types.str "sp color";
+
+    bold = helpers.mkNullOrOption types.bool "enable bold";
+
+    italic = helpers.mkNullOrOption types.bool "enable italic";
```
Comment (traxys) [inline plugins/bufferlines/bufferline.nix:22]: Aren't we missing the mkRenamedOptionModule ? Or are the options completely deleted from upstream ?

Comment (GaetanLepage) [reply plugins/bufferlines/bufferline.nix:22]: Finally done !

File: plugins/bufferlines/bufferline.nix [hunk 2]
```diff
@@ -10,25 +10,121 @@ with lib; let
...
       helpers.extraOptionsOptions
       // {
         enable = mkEnableOption "bufferline";
 
+        a = {
+          b = {
+            new = mkOption {};
+          };
+        };
+
```
Comment (traxys) [inline plugins/bufferlines/bufferline.nix:127]: This is a debugging part ?

File: plugins/bufferlines/bufferline.nix [hunk 3]
```diff
@@ -79,7 +175,8 @@ in {
           helpers.defaultNullOpts.mkStr "null"
           "Command or function run when middle clicking on a buffer.";
 
-        indicator = helpers.mkCompositeOption "Indicator" {
+        # indicator = helpers.mkCompositeOption "Indicator" {
```
Comment (traxys) [inline plugins/bufferlines/bufferline.nix:178]: Same ?

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/bufferlines/bufferline.nix (hunk 3 above)
Comment (GaetanLepage) [reply plugins/bufferlines/bufferline.nix:172]: Indeed

File: plugins/bufferlines/bufferline.nix (hunk 2 above)
Comment (GaetanLepage) [reply plugins/bufferlines/bufferline.nix]: Indeed

**Review by traxys — APPROVED** [review]

**Pull request merged**